### PR TITLE
[server] pausable

### PR DESCRIPTION
**Pull request by Quezler**
> prototype "it just sorta works" (code not deemed worthy to be merged)

- adds the ability for servers to be paused
- [ ] add server console (un)pause command?

### CODE REVIEW

**Comment by ZkyLB** [comment]
you scare me

**Comment by Quezler** [comment]
[this](https://github.com/Quezler/mindustry__plugin--weepingangel/blob/master/src/main/java/weepingangel/WeepingangelMod.java#L14) scares me more 👀

**Comment by ZkyLB** [comment]
> [this](https://github.com/Quezler/mindustry__plugin--weepingangel/blob/master/src/main/java/weepingangel/WeepingangelMod.java#L14) scares me more 👀

no you scare me more, you cute roomba

**Comment by Quezler** [comment]
krater

**Comment by CA6** [comment]
oh no........ tho quezler you're making things that are pretty cool keep it up

**Comment by Quezler** [comment]
thx ❤️

File: core/src/mindustry/core/NetClient.java [hunk 1]
```diff
@@ -50,6 +50,8 @@
     /** Last sent client snapshot ID. */
     private int lastSent;
 
+    public boolean paused = false;
```
Comment (Anuken) [inline core/src/mindustry/core/NetClient.java]: I'm concerned about the fact that there are now *4* different states to check for the pausing (some of which may be accurate, some may not): `netClient.paused`, `netServer.paused`, `state.isPaused()` and `state.is(State.paused)`. This is rather confusing, to say the least.

Comment (Quezler) [reply core/src/mindustry/core/NetClient.java]: well i wasn't sure where to put them and how to properly sync them all 🙈

**Comment by Anuken** [comment]
Since entities are now updated with a single call and there's no update delay, this will be easier to implement in 6.0. Will consider it there.

**Pull request closed without merging**